Haskell/MaîtreDTests.hs

```diff
@@ -0,0 +1,62 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module MaîtreDTests
+    ( tryAcceptBehavesCorrectlyWhenItCanAccept
+    , tryAcceptBehavesCorrectlyWhenItCanNotAccept
+    ) where
+
+import           Control.Monad      (liftM2)
+import           Data.Maybe         (isNothing)
+import           Data.Time          (ZonedTime (..), defaultTimeLocale,
+                                     iso8601DateFormat, parseTimeOrError)
+import           Data.Time.Calendar (gregorianMonthLength)
+import           MaîtreD
+import           Test.QuickCheck
+import           Text.Printf        (printf)
+
+instance Arbitrary ZonedTime where
+  arbitrary = do
+    y <- choose (1, 9999)
+    m <- choose (1, 12)
+    d <- choose (1, gregorianMonthLength y m)
+    return $
+      (parseTimeOrError True defaultTimeLocale $ iso8601DateFormat Nothing)
+      (printf "%04d" y ++ "-" ++ printf "%02d" m ++ "-" ++ printf "%02d" d)
```

Comment:
Is there any reason to attempt to _parse_ a date string, since we already have the values?

You can create a `ZonedTime` value from `y`, `m`, and `d` like this:

```haskell
ZonedTime (LocalTime (fromGregorian y m d) midnight) utc
```